### DISCUSSION TITLE
Status reporting

### DISCUSSION THREAD

**Discussion by regpye** [arkypita/LaserGRBL#1385] [Ideas]
Is there anyway to show the status of GRBL, for instance how can I see if a limit switch has been triggered?
I have set $10=19 but not been able to see anything when entering ?
This would make a nice feature if able to be done.

Also is there anyway to show more buttons in LaserGRBL? Maybe it could be possible to have a floating button panel that could be expanded?

**Comment by arkypita** [comment]
Hi

> Is there anyway to show the status of GRBL, for instance how can I see if a limit switch has been triggered?

There is a problem in showing the status report in plain text, it is that it is sent by the controller dozens of times per second.
If shown in the command console would scroll everything.

Instead of showing it LaserGRBL "intercepts" and parse it and shows its most significant values: the machine status (idle, alarm ...) and the override values at the bottom right. At the top right the current position, the current speed and power

I have not implemented any displays for the state of the limit switches, but if I wanted I could. In reality I don't think it is essential as, once you have verified that they work, you don't need to see it on the video whether they are pressed or not.

> Also is there anyway to show more buttons in LaserGRBL? Maybe it could be possible to have a floating button panel that could be expanded?

I can think to a double-size button bar but my question is: why did you need so much buttons? Framing, focusing, fan... and what more?
